### PR TITLE
Allow unanimatedProps passed through Axis.js to support not interpola…

### PR DESCRIPTION
**Pull request by CharlieMartell**
Previously, if you had an XYPlot, it would render its axis with hardcoded "animatedProps" meaning if you had anything in your axis-titles they would be ran through the d3-interpolator which is not always intended. 

This PR allows you to pass unanimatedProps and not interpolate some props of an axis via a XYPlot.

### CODE REVIEW

**Comment by CharlieMartell** [comment]
That makes sense I hadn't seen the PropTypes for animation being oneOfType allowing for nonAnimatedProps to be contained within animation.

File: src/plot/axis/axis.js [hunk 1]
```diff
@@ -137,9 +137,14 @@ class Axis extends PureComponent {
 
   render() {
 
-    const {animation} = this.props;
+    const {animation, unanimatedProps} = this.props;
+
 
     if (animation) {
+      const animatedProps = animation.nonAnimatedProps ? defaultAnimatedProps.filter(
+        prop => animation.nonAnimatedProps.indexOf(prop) < 0
```
Comment (mcnuttandrew) [inline src/plot/axis/axis.js]: could use find instead of indexof

File: docs/markdown/animation.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Animation makes your charts feel physical, it makes them feel alive, shoot it ma
 
 <!-- INJECT:"AnimationExample" -->
 
-*Objects*: react-motion expects an object formatting like {damping: NUMBER, stiffness: NUMBER}, and if you want to give us an object like that, we will hand it direct to react-motion.
+*Objects*: React-motion expects an object formatting like `{damping: NUMBER, stiffness: NUMBER}`, and if you want to give us an object like that, we will hand it direct to react-motion. You can also pass an object with `{nonAnimatedProps: ['foo', 'bar']}` to prevent those props from being interpolated by d3-interpolator.
```
Comment (mcnuttandrew) [inline docs/markdown/animation.md]: Nb I think they spell it lowercase

File: src/plot/xy-plot.js [hunk 3]
```diff
@@ -305,8 +304,8 @@ class XYPlot extends React.Component {
         dataProps = {data: data[seriesIndex]};
       }
       return React.cloneElement(child, {
-        ...dimensions,
         animation,
+        ...dimensions,
```
Comment (mcnuttandrew) [inline src/plot/xy-plot.js]: Why the reorder?

File: docs/markdown/animation.md (hunk 2 above)
Comment (CharlieMartell) [inline docs/markdown/animation.md]: Ah I was just trying to make it the same as in +*String*+, I will go ahead and change both.

**Pull request merged**